### DISCUSSION TITLE
Is there a way to import Base for table creation without first having to import all the classes that inherit from it?

### DISCUSSION THREAD

**Discussion by jalvespinto** [sqlalchemy/sqlalchemy#7906] [Usage Questions]
I have an app where I declare a `Base = declarative_base()` in one file and the uses it in many different classes living in different modules. Now I am writing a manage script that, among other things, I use to create tables using `Base.metadata.create_all`. The script works, but I have to import all the modules that have classes that inherit from `Base`. Is this the only way or is there a away to avoid all those imports?

**Comment by zzzeek** [comment]
It's by definition not possible for ``create_all()`` to emit the CREATE statements for Table objects that would be defined in Python, if they were not in fact loaded into memory.    If you have a large application with lots of classes, you might find it more appropriate to use [alembic migrations](https://alembic.zzzcomputing.com/) so that the structure of your schema can be rendered out to the database independently of your model files.
